internal/pkg/config/config.go

```diff
@@ -168,6 +169,29 @@ func redactOutput(cfg *Config) Output {
 		redacted.Elasticsearch.TLS = &newTLS
 	}
 
+	if redacted.Elasticsearch.Headers != nil {
+		headers := make(map[string]string)
+		for k, v := range redacted.Elasticsearch.Headers {
+			headers[k] = v
+			lk := strings.ToLower(k)
+			if strings.Contains(lk, "auth") || strings.Contains(lk, "token") || strings.Contains(lk, "key") || strings.Contains(lk, "bearer") { // best-effort scan to redact sensitive headers
+				headers[k] = kRedacted
+			}
+		}
+		redacted.Elasticsearch.Headers = headers
+	}
+
+	if redacted.Elasticsearch.ProxyHeaders != nil {
+		proxyHeaders := make(map[string]string)
+		for k, v := range redacted.Elasticsearch.ProxyHeaders {
+			proxyHeaders[k] = v
+			lk := strings.ToLower(k)
+			if strings.Contains(lk, "auth") || strings.Contains(lk, "token") || strings.Contains(lk, "key") || strings.Contains(lk, "bearer") { // best-effort scan to redact sensitive headers
```

Comment:
This should probably be a function just to prevent people from forgetting to update both places.

We could also one day split out a lot of our best effort redaction logic into a reusable package.